logstash/ruby/flat-source.rb

```diff
@@ -0,0 +1,21 @@
+def register(params)
+  @source = params["source"]
+end
+
+# Move reminder properties to the root-level
+def filter(event)
+  # tag @source_field_not_found when source fields is not present
+  if event.get(@source).nil?
+    event.tag("#{@source}_not_found")
+    return [event]
+  end
+
+  event
+    .get(@source)
+    .each do |k, v|
+      event.set(k, v)
+  end
```

Comment:
Small style complain: `end` looks like it ends a block begin by `event`, but this is `do-end` block, so `end` should be indented.